clients/fides-js/src/lib/tcf.ts

```diff
@@ -39,17 +39,22 @@ export const generateTcString = async ({
     tcModel.cmpVersion = CMP_VERSION;
     tcModel.consentScreen = 1; // todo- On which 'screen' consent was captured; this is a CMP proprietary number encoded into the TC string
 
+    // Narrow the GVL to say we've only showed these vendors provided by our experience
+    const vendorIds = [
+      ...(experience.tcf_vendor_consents?.map((v) => +v.id) || []),
+      ...(experience.tcf_vendor_legitimate_interests?.map((v) => +v.id) || []),
+    ];
+    tcModel.gvl.narrowVendorsTo(vendorIds);
+
     if (tcStringPreferences) {
-      if (
-        tcStringPreferences.vendorsConsent &&
-        tcStringPreferences.vendorsConsent.length > 0
-      ) {
```

Comment:
I don't think we need these `if`s: it's always an array, and even if it's empty, the `forEach` below won't do anything